### ISSUE TITLE
Hot Reload not working

### ISSUE THREAD

**Issue by danardelean** [microsoft/vscode-dotnettools#379]
Type: <b>Bug</b>

Is hot reload supposed to work with the released or preview version of this plugin?

Extension version: 0.3.1
VS Code version: Code 1.81.1 (6c3e3dba23e8fadc360aed75ce363ba185c49794, 2023-08-09T22:40:25.698Z)
OS version: Darwin arm64 22.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|1, 3, 2|
|Memory (System)|16.00GB (0.07GB free)|
|Process Argv|--crash-reporter-id 84b02e1c-434b-4987-83d0-836b0bac9286|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
vscaat:30438848
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
vsccc:30803844
282f8724:30602487
f6dab269:30613381
vscrpc:30673769
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
57b77579:30736110
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofb:30804716
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
dsvsc013:30795093
dsvsc014:30804076

```

</details>

**Comment by rogersm-uwosh** [comment]
Was this addressed? I really, really miss hot reload!

**Comment by siyavash1984** [comment]
This Bug and question has been opened for 3 weeks, is there any update or roadmap for this feature?

**Comment by poppastring** [comment]
Hi @danardelean @rogersm-uwosh @siyavash1984 

We added support for Edit and Continue (EnC) and Hot Reload when debugging managed applications with the C# dev kit preview. Please note that EnC/Hot Reload debugging on Mac and Linux requires .NET 8 RC1 and above. In order to start using this feature in the prerelease, enable the **csharp.debug.hotReloadOnSave** setting.

Please check it out and give us feedback!

**Comment by isadorasophia** [comment]
Closing this issue, please feel free to open a new one if you have any issues using hot reload!

**Comment by rogersm-uwosh** [comment]
Thank you so much for doing this, I can't wait to use it. 

Just to confirm, on Windows this will work on .NET 7? I'm at university, and IT will not install beta software (for very good reason). I'm on a Mac, and have no problems upgrading (assuming that .NET 7 and 8 can coexist).

**Comment by mprogers** [comment]
One last question - are there instructions on how to work with both .NET 7 and .NET 8 in Visual Studio Code? I know in Visual Studio for Mac, you just select it from the pull down when creating a new project.

**Comment by danardelean** [comment]
@poppastring Tried on macOS and it does not work. 

I have even enabled CSharp>Experimental>Debug>Hot Reload (you only said Hot Reload On Save which was true by default)
The project is an .NET MAUI net8.0. 
C# Dev Kit v0.5.98 
.NET MAUI v0.4.3
8.0.100-rc.1.23463.5 [/usr/local/share/dotnet/sdk]
 Microsoft.NETCore.App 8.0.0-rc.1.23419.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
 macOS Sonoma 14.0

**Comment by poppastring** [comment]
@danardelean I just spotted an error in my original response. The setting to update is **csharp.experimental.debug.hotReload**. Could you try that again and ping me if you have an issue?

**Comment by isadorasophia** [comment]
> are there instructions on how to work with both .NET 7 and .NET 8 in Visual Studio Code?

@mprogers You can change the .csproj so it targets either .net 7 or .net 8:
```xml
<TargetFramework>net8.0</TargetFramework>
```

or creating a new project with:
```ps1
dotnet new console -f=net7.0
```

I'm not sure if there's an actual UI for that in C# dev kit. You can open a new issue if you have any suggestions on making that more intuitive.

> The project is an .NET MAUI net8.0.

@danardelean C# dev kit doesn't support hot reload for .NET MAUI apps yet, unfortunately

**Comment by rogersm-uwosh** [comment]
This is hard to follow - you've got hot reload working on .NET 7 for Windows, .NET 8rc1 for macOS, but it does not support .NET MAUI? Have I got that all correct? [I actually hope not, because I'm working with .NET MAUI on macOS.]

**Comment by poppastring** [comment]
@rogersm-uwosh Yes, you have summarized it perfectly :), I will repeat here for folks following this issue:

- C# dev kit supports Hot Reload on Windows (requires .NET 7 or above) and MacOS (requires .NET 8 RC1 or above)
- C# dev kit does not support Hot Reload for .NET MAUI apps.

**We are actively working on Hot Reload support for .NET MAUI**, but I do not have a release date to share today.

**Comment by rogersm-uwosh** [comment]
Thanks so much, Mark - the words “actively working on” give me hope 😀

Michael
===============
Dr. Michael Rogers
Assistant Professor, Computer Science
University of Wisconsin Oshkosh
Office: 214 Halsey Science Center
Social Media (Twitter/FB/Instagram): mprogers
Text/Phone: 309-825-6454
Hobbies: Angel Flights, Crossword Solving
Podcast: Jean & Mike Do The New York Times Crossword<https://jamcr.buzzsprout.com/>
Pronouns: he, him, his





On Sep 28, 2023, at 2:07 PM, Mark Downie ***@***.***> wrote:


CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.




@rogersm-uwosh<https://github.com/rogersm-uwosh> Yes, you have summarized it perfectly :), I will repeat here for folks following this issue:

  *   C# dev kit supports Hot Reload on Windows (requires .NET 7 or above) and MacOS (requires .NET 8 RC1 or above)
  *   C# dev kit does not support Hot Reload for .NET MAUI apps.

We are actively working on Hot Reload support for .NET MAUI, but I do not have a release date to share today.

**Comment by bronius** [comment]
> Hot Reload support for .NET MAUI

Is there [an open issue on Hot Reload support for .NET MAUI](https://github.com/microsoft/vscode-dotnettools/issues?q=is%3Aissue+is%3Aopen+Hot+Reload+support+for+.NET+MAUI) one can track to follow development?

**Comment by dineshmike** [comment]
> > We are actively working on Hot Reload support for .NET MAUI, but I do not have a release date to share today.
> 
> Is there [an open issue on Hot Reload support for .NET MAUI](https://github.com/microsoft/vscode-dotnettools/issues?q=is%3Aissue+is%3Aopen+Hot+Reload+support+for+.NET+MAUI) one can track to follow development?

#867

**Comment by sirj0k3r** [comment]
I'd like to know if there's anyone currently with an issue where **Hot Reload** seems to be working but simply states that "error **ENC1005**: The current content of source file does not match the built source. Any changes made to this file while debugging won't be applied until its content matches the built source." Rendering it a bit unhelpful.

I'm using **.NET 8** and on a **macOS** in which **Hot Reload** had previously worked with a similar project.

I also notice that the `Show all dynamic configurations` button is not showing under the Run & Debug side menu, despite having **C# Dev Kit** installed and a few profiles under `launchSettings.json`
